js/src/common/components/Button.tsx

```diff
@@ -5,46 +5,64 @@ import extractText from '../utils/extractText';
 import LoadingIndicator from './LoadingIndicator';
 
 export interface ButtonProps extends ComponentProps {
+    /**
+     * A tooltip for the button.
+     */
     title?: string;
+
+    /**
+     * An html type attribute for the button.
+     */
     type?: string;
+
+    /**
+     * The name of the icon class. If specified, the button will be given a
+     * 'has-icon' class name.
+     */
     icon?: string;
 
+    /**
+     * Whether or not the button should be in a disabled loading state.
+     */
     loading?: boolean;
+
+    /**
+     * Whether or not the button is disabled. If truthy, the button
+     * will be given a 'disabled' class name, and any `onclick` handler will be
+     * removed.
+     */
     disabled?: boolean;
+
+    /**
+     * A callback to run when the button is clicked.
+     */
     onclick?: Function;
 }
 
 /**
  * The `Button` component defines an element which, when clicked, performs an
- * action. The button may have the following special props:
- *
- * - `icon` The name of the icon class. If specified, the button will be given a
- *   'has-icon' class name.
- * - `disabled` Whether or not the button is disabled. If truthy, the button
- *   will be given a 'disabled' class name, and any `onclick` handler will be
- *   removed.
- * - `loading` Whether or not the button should be in a disabled loading state.
- *
- * All other props will be assigned as attributes on the button element.
+ * action.
  *
  * Note that a Button has no default class names. This is because a Button can
  * be used to represent any generic clickable control, like a menu item.
  */
-export default class Button<T extends ButtonProps = ButtonProps> extends Component<T> {
+export default class Button<T extends ButtonProps = ButtonProps> extends Component<ButtonProps> {
     view() {
-        const { children, ...attrs } = this.props;
+        const attrs: ButtonProps = { ...this.props };
```

Comment:
This *could* use `T`, not necessary though because you are only using it in `view`, but perhaps for consistency? Doesn't matter to me, just some thoughts.